[Sean Chin] iP

seannzx opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _**SUPER**_ **FAST** to use

All you need to do is,

1. Download it from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/153)
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you 😋 

And it is **FREE**!

Features: 
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seannzx committed:
Level-1

seannzx committed:
Level-2

seannzx committed:
Level-3

seannzx committed:
Level-4

seannzx committed:
A-TextUiTesting

seannzx committed:
Level-5

seannzx committed:
Level-6

seannzx committed:
Edit switch case indentations

seannzx committed:
Create new function to print error messages based on type

seannzx committed:
Add ability to read and access tasklist.txt file

seannzx committed:
Fixed a bug in parseTaskFromFile() function

seannzx committed:
Level-7

seannzx committed:
Level-8

seannzx committed:
Cleaned up some code placement

seannzx committed:
Merge branch 'branch-Level-7'

seannzx committed:
Merged with branch-Level-8

seannzx committed:
Redesign DUKE with more OOP concepts.

seannzx committed:
Move printListForSave() in UI class to STORAGE class

seannzx committed:
Package DUKE into various files

seannzx committed:
Add JUnit test for Commands

seannzx committed:
Change project directory in Storage class

seannzx committed:
Update all files to latest version

seannzx committed:
Add JavaDoc Comments for Duke Class

seannzx committed:
Add JavaDoc Comments for Parser Class

seannzx committed:
Add JavaDoc Comments for Storage Class

seannzx committed:
Add JavaDoc Comments for TaskList Class

seannzx committed:
Add JavaDoc Comments for Ui Class

seannzx committed:
Add JavaDoc Comments for all the Commands

seannzx committed:
Add JavaDoc Comments for all Tasks

seannzx committed:
Add JavaDoc Comments for all Exceptions

seannzx committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc' into branch-A-JavaDoc

seannzx committed:
Edit classes to follow the Java coding standard

seannzx committed:
ADd support to find keywords

seannzx committed:
Merge branch 'branch-Level-9'

seannzx committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

seannzx committed:
Update JavaDoc methods for classes in the main Duke package

seannzx committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java

seannzx committed:
Merge remote-tracking branch 'origin/add-gradle-support'

seannzx committed:
Add checkstyle support

seannzx committed:
Add GUI support to Duke

seannzx committed:
Add CheckStyle updates

seannzx committed:
Add CheckStyle updates in DialogBox & ListCommand

Chang-CH commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * A launcher class to workaround classpath issues.
+ */
+
```
Redundant newline here

Chang-CH replied:
newline can be removed so readers can easily tell the javadoc is for the Launcher class

Chang-CH commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,63 @@
+        scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+    }
+
+    public void setDuke(Duke d) {
```
Javadocs can be added here

Chang-CH commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    /** Helper function to determine if the specified path ends with .txt */
```
@param tags can be added to make the javadocs more informative

Chang-CH commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    /**
+     * Adds the specified task to the tasklist and reflects the result via the Ui.
```
General comment should be separated from tags by a newline

Chang-CH commented on src/main/java/duke/commands/DoneCommand.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Helper function to tell Duke to continue reading inputs
+     */
+    public boolean isExit() {
```
checkIsExit might be a better name, alternatively make isExit a static boolean field instead.

Chang-CH reviewed (commented):
Overall, I found your code easy to read for the most part. I noted a few minor coding standard violations too. LGTM, almost. Just a few nits to fix. 👍

Chang-CH commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,133 @@
+        return output;
+    }
+
+    /** Helper function to get the first word of the input sentence */
```
javadocs for methods should include parameters

Chang-CH commented on src/main/java/duke/commands/AddCommandTest.java:
```diff
@@ -0,0 +1,40 @@
+        Assert.assertEquals(add.getTaskDescription(), description);
+    }
+
+    @Test
```
I like how you have tests for individual task types to ensure reliability 👍

Chang-CH reviewed (commented):
As continued from previous review

seannzx committed:
Minor edits

seannzx committed:
Add assertions to certain classes

seannzx committed:
Add assertions to certain classes

Assertion adds extra levels of checks for code logic

For functions which modify the tasklist, there is a need to ensure that the specified index to retrieve the task from is valid.

As the program utilises relative file paths to deal with information,
 there must be a level of check to ensure that the file path is valid

seannzx committed:
Add assertions to certain classes

Assertion adds extra levels of checks for code logic

For functions which modify the tasklist, there is a need to ensure that
the specified index to retrieve the task from is valid

As the program utilises relative file paths to deal with information,
there must be a level of check to ensure that the file path is valid

seannzx committed:
Merge remote-tracking branch 'origin/A-Assertions' into A-Assertions

seannzx committed:
Merge remote-tracking branch 'origin/A-Assertions' into A-Assertions

seannzx committed:
Improve code quality in Storage class

The function convertToTask() parses text from the storage file by
checking the input length. As the text format is fixed, defining a
helper function to check the validity of the format is more intuitive

seannzx committed:
Merge pull request #1 from xnajasho/A-Assertions

A-assertions

seannzx committed:
Merge branch 'master' into A-CodeQuality

seannzx committed:
Merge pull request #2 from xnajasho/A-CodeQuality

Improve code quality in Storage class

seannzx committed:
Add support for DoWithinPeriod tasks

seannzx committed:
Update README.md

seannzx committed:
Update README.md 2.0

seannzx committed:
Shorten Welcome Message

seannzx committed:
Add Ui.png
